scripts/bfasst/compare_waveforms/Templates/get_paths.py

```diff
@@ -1,42 +1,84 @@
-import bfasst
-from bfasst.compare_waveforms.Templates import structs
+paths = {}
 
-paths = structs.paths
 
-def get_paths(self, design):
-    paths["build_dir"] = self.work_dir
-    paths["path"].append(design.impl_netlist_path)
-    paths["path"].append(design.reversed_netlist_path)
-    paths["file"].append(paths["build_dir"] / (paths["path"][0].name))
-    paths["file"].append(paths["build_dir"] / (paths["path"][1].name))
-    paths["modules"].append(
-        paths["path"][0].name[0 : len(paths["path"][0].name) - 7]
-    )
-    paths["modules"].append(
-        paths["path"][0].name[0 : len(paths["path"][0].name) - 2]
-    )
-    paths["modules"].append(
-        paths["path"][1].name[0 : len(paths["path"][1].name) - 2]
-    )
-    paths["diff"] = paths["build_dir"] / "diff.txt"
-    paths["tb"].append(paths["build_dir"] / (paths["modules"][1] + "_tb.v"))
-    paths["tb"].append(paths["build_dir"] / (paths["modules"][2] + "_tb.v"))
-    paths["vcd"].append(paths["build_dir"] / (paths["modules"][1] + ".vcd"))
-    paths["vcd"].append(paths["build_dir"] / (paths["modules"][2] + ".vcd"))
-    paths["temp_vcd"].append(paths["build_dir"] / (paths["modules"][1] + "_temp.vcd"))
-    paths["temp_vcd"].append(paths["build_dir"] / (paths["modules"][2] + "_temp.vcd"))
-    paths["dsn"] = paths["build_dir"] / ("dsn")
-    paths["cells_sim"] = bfasst.paths.ROOT_PATH / (
-        "third_party/yosys/techlibs/xilinx/cells_sim.v"
-    )
-    paths["sample_tb"] = bfasst.paths.ROOT_PATH / (
-        "scripts/bfasst/compare_waveforms/Templates/sample_tb.v"
-    )
-    paths["tcl"].append(paths["build_dir"] / (paths["modules"][1] + ".tcl"))
-    paths["tcl"].append(paths["build_dir"] / (paths["modules"][2] + ".tcl"))
-    paths["fst"].append(paths["build_dir"] / (paths["modules"][1] + "_temp.vcd.fst"))
-    paths["fst"].append(paths["build_dir"] / (paths["modules"][2] + "_temp.vcd.fst"))
-    paths["parsed_diff"] = self.work_dir / ("parsed_diff.txt")
-    paths["test"] = paths["build_dir"] / "test.v"
+def get_paths(work_dir, root_path, design):
+    # The base directory that files are stored in
+    paths.setdefault("build_dir", work_dir)
```

Comment:
Sorry, I was a little unclear.  setdefault is used when accessing a dictionary item, to say get the item if it exists, otherwise return this value.  It doesn't change anything inherently in the dictionary.  I only suggested it because it seemed that you were attempting to avoid appending to a Null value, in the case of a list.  For example, instead of doing something like this:
```python
a = {"paths": []}
a["paths"].append("/usr/local/bin")
```
You can do this:
```python
a = {}
a.setdefault("paths", []).append("/usr/local/bin")
```
and you don't have to worry about appending to null.

Reply:
Although, in this case, it might be easier to just change this:
```python
    paths["path"].append(design.impl_netlist_path)
    paths["path"].append(design.reversed_netlist_path)
```
to this:
```python
    paths["path"] = []
    paths["path"].append(design.impl_netlist_path)
    paths["path"].append(design.reversed_netlist_path)
```
That's probably the cleanest to me.